solana/restaking/programs/restaking/src/lib.rs

```diff
@@ -68,13 +73,76 @@ pub mod restaking {
             ctx.accounts.token_mint.key()
         );
 
-        staking_params
+        let token_index = staking_params
             .whitelisted_tokens
             .iter()
-            .find(|&&token_mint| token_mint == ctx.accounts.token_mint.key())
+            .enumerate()
+            .find_map(|(index, &token_mint)| {
+                if token_mint == ctx.accounts.token_mint.key() {
+                    return Some(index);
+                }
+                None
+            })
             .ok_or_else(|| error!(ErrorCodes::TokenNotWhitelisted))?;
 
-        staking_params.total_deposited_amount += amount as u128;
+        let token_price_update = &ctx.accounts.token_price_update;
+        let sol_price_update = &ctx.accounts.sol_price_update;
+
+        let sol_price_feed_id = SOL_PRICE_FEED_ID.to_string();
```

Comment:
Why convert it to String?

Reply:
because the feed id from hex expects &string and if i convert it there then the compiler complains saying use of moved value.

Reply:
I think what you want is:

```rust
        let token_feed_id = staking_params
            .token_oracle_addresses
            .get(token_index)
            .as_deref()
            .unwrap_or(SOL_PRICE_FEED_ID);
```

Reply:
oh ya i think `deref` is what was needed.